requests/models.py

```diff
@@ -414,7 +414,7 @@ def prepare_body(self, data, files, json=None):
         content_type = None
         length = None
 
-        if json is not None:
+        if data == {} and json is not None:
```

Comment:
Skeptical of the comparison? If yes, following is a suggestion
if not data and json is not None:
    ....